app/controllers/files_controller.rb

```diff
@@ -109,4 +104,45 @@ def filename
   def load_resource
     @cocina_model = Repository.find(params[:item_id])
   end
+
+  # Zip-tricks based streaming for files from preservation.
+  # Based on https://piotrmurach.com/articles/streaming-large-zip-files-in-rails/
+  class PresStreamer
```

Comment:
It seems like one of the advantages of extracting this functionality as a class is that it can be tested individually? Maybe not though...

Reply:
I was thinking the same, and wondering about giving this class its own file (in `app/services/` or something, but I'm also pondering whether testing this class in isolation might be more trouble than it's worth because:

1. Most of what's in the class is ZipTricks implementation details which---one hopes, lol---is already being tested by the ZipTracks maintainers.
2. What's not ZipTricks implementation details is in a fairly nested block structure which may make for difficult testing.

I'm on the fence but can be convinced either way.

Reply:
I opted not to test in more detail for the reasons that @mjgiarlo gave. It didn't seem that additional testing would give greater assurance that "downloading all" actually worked.